beets/util/artresizer.py

```diff
@@ -232,23 +243,33 @@ def _check_method():
 
 def get_im_version():
     """Return Image Magick version or None if it is unavailable
-    Try invoking ImageMagick's "convert".
+    Try invoking ImageMagick's "magick". If "magick" is unavailable,
+    as with older versions, fall back to "convert"
+
+    Our iterator `im_legacy` will be non-zero when the first command
+    fails, and will be returned in a tuple along with the version
     """
-    try:
-        out = util.command_output(['convert', '--version'])
-
-        if b'imagemagick' in out.lower():
-            pattern = br".+ (\d+)\.(\d+)\.(\d+).*"
-            match = re.search(pattern, out)
-            if match:
-                return (int(match.group(1)),
-                        int(match.group(2)),
-                        int(match.group(3)))
-            return (0,)
-
-    except (subprocess.CalledProcessError, OSError) as exc:
-        log.debug(u'ImageMagick check `convert --version` failed: {}', exc)
-        return None
+    cmds = ('magick', 'convert')
+    for im_legacy, cmd in enumerate(cmds):
+
+        try:
+            out = util.command_output([cmd, '--version'])
+
+            if b'imagemagick' in out.lower():
+                pattern = br".+ (\d+)\.(\d+)\.(\d+).*"
+                match = re.search(pattern, out)
+                if match:
+                    return ((int(match.group(1)),
+                            int(match.group(2)),
+                            int(match.group(3))),
+                            bool(im_legacy)
+                            )
+
+        except (subprocess.CalledProcessError, OSError) as exc:
+            log.debug(u'ImageMagick version check failed: {}', exc)
+            return None
+
+        return (0,)
```

Comment:
The return type of this function could be a little cleaner and more regular: perhaps it should _always_ return a pair of a "legacy" flag (a Boolean) and a version. The return value here doesn't include the flag.

Reply:
Added in 48be3a7

Reply:
I forgot a set of parentheses here and returned `(0, None)` rather than `((0,), None)`. After thinking about it some more though, I wonder if it should be reverted rather than fixed.

We know that if the version returns 0, all usable commands have failed and we can safely assume that ImageMagick is not installed or accessible. In this case, the legacy flag doesn't make sense since we a) have no installation to refer to as legacy or not, and b) can't make use of ImageMagick anyway. 

Since all existing code that checks the version refers to it using `self.method[1]`, we'd then have to change each of these to `self.method[1][0]` or else explicitly store the version in another variable. Doesn't seem worth it just to return a flag that will never be used.

I'm going to revert this specific change for now, but let me know your thoughts.